internal/pkg/plugin/binary.go

```diff
@@ -115,24 +115,38 @@ func InstallFromSIF(fimg *sif.FileImage, libexecdir string) (*Meta, error) {
 // Uninstall removes the plugin matching "name" from the specified
 // singularity installation directory
 func Uninstall(name, libexecdir string) error {
-	sylog.Debugf("Uninstalling plugin %q from %q", name, libexecdir)
-
-	metas, err := List(libexecdir)
+	pluginDir := filepath.Join(libexecdir, DirRoot)
+	sylog.Debugf("Uninstalling plugin %q from %q", name, pluginDir)
 
+	meta, err := loadMetaByName(name, pluginDir)
 	if err != nil {
+		// figure out if this is a not found error?
```

Comment:
+1 here, let's tackle this in a follow up PR. I think there is actually a decent chunk of useful work we can do with plugins error. In `pkg/plugin` we can add a `pkg/plugin/error.go` or similar where we define many `pluginapi` specific errors.